### DISCUSSION TITLE
Does Command Guard Properties need to raise property changed event?

### DISCUSSION THREAD

**Discussion by mwdpb** [canton7/Stylet#213] [Q&A] [answered]
Please see the viewmodel snippet here.  The ```Test``` method gets invoked when I click the button, however, the ```CanTest``` is set to false in the constructor, therefore the button should've been disabled.  Do I need raise property changed event in order for it to work?  Thanks!

![image](https://user-images.githubusercontent.com/55845355/107779466-61f22f00-6d13-11eb-9b95-16cbdc39f898.png)

Here is the xaml snippet:

![image](https://user-images.githubusercontent.com/55845355/107779531-76cec280-6d13-11eb-891d-68c5f7a23f78.png)

**Comment by mwdpb** [comment, marked as answer]
Just tried raising the property changed event and it works.  Maybe update the manual example?  Otherwise newbies might have trouble to get it to work.

![image](https://user-images.githubusercontent.com/55845355/107780514-c5c92780-6d14-11eb-9c01-27861359377f.png)

Here is the section in manual talking about the guard property, unless I'm mistaken, it doesn't work.

![image](https://user-images.githubusercontent.com/55845355/107781420-e2199400-6d15-11eb-9a20-93fc3ec4e55a.png)

**Reply by canton7** [reply]
Ah yes, it looks like the action never bothers to check for guard properties unless the target implements `INotifyPropertyChanged`, [see here](https://github.com/canton7/Stylet/blob/b2db34a8014efee0bd23d03e965fbc901e6038a4/Stylet/Xaml/CommandAction.cs#L74). 

My current gut feeling is that we probably should poll `CanTest` once if the target doesn't implement `INotifyPropertyChanged`: that supports the case where someone wants to disable a button based on a condition in the ctor, but doesn't implement INPC for some reason. Thoughts?

I've updated the wiki page on Actions.

**Reply by mwdpb** [reply]
Agree.  Maybe do a warning like you did for bool type check below?  However, you can't even disable a button in the ctor unless it is INPCed.  So I'd say we check it anyhow, the case you described should be a business logic instead of a framework logic.  

```Found guard property {0} for action {1} on target {2}, but its return type wasn't bool. Therefore, ignoring", this.GuardName, this.MethodName, newTarget```

**Reply by canton7** [reply]
See #214
